components/match2/wikis/warcraft/match_group_input_custom.lua

```diff
@@ -0,0 +1,776 @@
+
+local Opponent = require('Module:OpponentLibraries').Opponent
+
+local ALLOWED_STATUSES = {'W', 'FF', 'DQ', 'L'}
+local CONVERT_STATUS_INPUT = {W = 'W', FF = 'FF', L = 'L', DQ = 'DQ', ['-'] = 'L'}
+local DEFAULT_LOSS_STATUSES = {'FF', 'L', 'DQ'}
+local MAX_NUM_OPPONENTS = 2
+local MAX_NUM_PLAYERS = 20
+local DEFAULT_BEST_OF = 99
+local LINKS_KEYS = {'preview', 'preview2', 'interview', 'interview2', 'review', 'recap', 'lrthread'}
+local MODE_MIXED = 'mixed'
+local TBD = 'tbd'
+local NEUTRAL_HERO_FACTION = 'neutral'
+
+local CustomMatchGroupInput = {}
+
+--- called from Module:MatchGroup
+---@param match table
+---@param options table?
+---@return table
+function CustomMatchGroupInput.processMatch(match, options)
+	assert(not Logic.readBool(match.ffa), 'FFA is not yet supported in warcraft match2')
+
+	Table.mergeInto(
+		match,
+		CustomMatchGroupInput._readDate(match)
+	)
+	match.patch = PatchAuto.retrieve{date = match.date}
+	CustomMatchGroupInput._getTournamentVars(match)
+	CustomMatchGroupInput._adjustData(match)
+	CustomMatchGroupInput._updateFinished(match)
+	match.stream = Streams.processStreams(match)
+	CustomMatchGroupInput._getLinks(match)
+	CustomMatchGroupInput._getExtraData(match)
+
+	return match
+end
+
+---@param matchArgs table
+---@return table
+function CustomMatchGroupInput._readDate(matchArgs)
+	if matchArgs.date then
+		local dateProps = MatchGroupInput.readDate(matchArgs.date)
+		dateProps.dateexact = Logic.nilOr(Logic.readBoolOrNil(matchArgs.dateexact), dateProps.dateexact)
+		Variables.varDefine('matchDate', dateProps.date)
+		return dateProps
+	end
+
+	local suggestedDate = Variables.varDefaultMulti(
+		'matchDate',
+		'Match_date',
+		'tournament_startdate',
+		'tournament_enddate',
+		'1970-01-01'
+	)
+
+	return {
+		date = MatchGroupInput.getInexactDate(suggestedDate),
+		dateexact = false,
+	}
+end
+
+---@param match table
+function CustomMatchGroupInput._updateFinished(match)
+	match.finished = Logic.nilOr(Logic.readBoolOrNil(match.finished), Logic.isNotEmpty(match.winner))
+	if match.finished then
+		return
+	end
+
+	-- Match is automatically marked finished upon page edit after a
+	-- certain amount of time (depending on whether the date is exact)
+	local currentUnixTime = os.time(os.date('!*t') --[[@as osdateparam]])
+	local matchUnixTime = tonumber(mw.getContentLanguage():formatDate('U', match.date))
+	local threshold = match.dateexact and 30800 or 86400
+	match.finished = matchUnixTime + threshold < currentUnixTime
+end
+
+---@param match table
+function CustomMatchGroupInput._getTournamentVars(match)
+	match.cancelled = Logic.emptyOr(match.cancelled, Variables.varDefault('cancelled tournament', 'false'))
+	match.headtohead = Logic.emptyOr(match.headtohead, Variables.varDefault('headtohead'))
+	Variables.varDefine('headtohead', match.headtohead)
+	match.publishertier = Logic.emptyOr(match.featured, Variables.varDefault('tournament_publishertier'))
```

Comment:
Is this correct (featured sets publishertier)?

Reply:
leftover from sc2^^
```suggestion
	match.publishertier = Logic.emptyOr(match.publishertier, Variables.varDefault('tournament_publishertier'))
```